persistence/trees/module.go

```diff
@@ -3,19 +3,26 @@ package trees
 import (
 	"fmt"
 
+	"github.com/pokt-network/pocket/persistence/indexer"
 	"github.com/pokt-network/pocket/persistence/kvstore"
 	"github.com/pokt-network/pocket/shared/modules"
 	"github.com/pokt-network/smt"
 )
 
-func (*treeStore) Create(bus modules.Bus, options ...modules.TreeStoreOption) (modules.TreeStoreModule, error) {
-	m := &treeStore{}
+var _ modules.Module = &TreeStore{}
+
+func (*TreeStore) Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
```

Comment:
Is there a reason to not return `modules.TreeStoreModule` here? Now submodules work properly?